PennMobile/Events/Views/PennEventsView.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  PennEventsView.swift
+//  PennMobile
+//
+//  Created by Jacky on 3/10/24.
+//  Copyright © 2024 PennLabs. All rights reserved.
+//
+
+import SwiftUI
+
+struct PennEventsView: View {
+    
+    @StateObject var viewModel = PennEventsViewModel()
+    
+    private var categories: [String] {
+        viewModel.uniqueEventTypes
+    }
+
+    var body: some View {
+//        NavigationView {
+            VStack {
+                Picker("Select Category", selection: $viewModel.selectedCategory) {
+                    ForEach(viewModel.uniqueEventTypes, id: \.self) { category in
+                        Text(category).tag(category)
+                    }
+                }
+                .pickerStyle(SegmentedPickerStyle())
+                .padding()
+                
+//                Menu {
+//                    ForEach(categories, id: \.self) { category in
+//                        Button(action: {
+//                            viewModel.selectedCategory = category
+//                        }) {
+//                            Text(category)
+//                        }
+//                    }
+//                } label: {
+//                    HStack {
+//                        Text(viewModel.selectedCategory)
+//                        Image(systemName: "chevron.down")
+//                    }
+//                    .foregroundColor(.accentColor)
+//                    .padding()
+//                    .overlay(
+//                        RoundedRectangle(cornerRadius: 10)
+//                            .stroke(Color.accentColor, lineWidth: 1)
+//                    )
+//                }
+//                .padding()
+
+                ScrollView {
+                    LazyVStack {
+                        ForEach(viewModel.selectedCategory == "All" ? viewModel.events : viewModel.events.filter { $0.eventType == viewModel.selectedCategory }, id: \.id) { event in
+                            NavigationLink(destination: PennEventsViewerView(event: event)) {
+                                PennEventCellView(viewModel: event)
+                            }
+                        }
+                    }
+                }
+            }
+            .onAppear {
+                viewModel.fetchEvents()
+            }
+//        }
+    }
+
+}
+
+struct PennEventsView_Previews: PreviewProvider {
```

Comment:
We can use the `#Preview` macro in future. Won't block on this but just good to know